[yang] Add yang model for POE_PORT

SerhiyBoikoPLV opened the pull request:
<!--
     Please make sure you've read and understood our contributing guidelines:
     https://github.com/Azure/SONiC/blob/gh-pages/CONTRIBUTING.md

     ** Make sure all your commits include a signature generated with `git commit -s` **

     If this is a bug fix, make sure your description includes "fixes #xxxx", or
     "closes #xxxx" or "resolves #xxxx"

     Please provide the following information:
-->

#### Why I did it

POE_PORT yang model is required for PoE CLI validation

HLD: https://github.com/sonic-net/SONiC/pull/1631

Related PRs:
- https://github.com/sonic-net/sonic-buildimage/pull/19636
- https://github.com/sonic-net/sonic-utilities/pull/3436

##### Work item tracking

#### How I did it

#### How to verify it

<!--
If PR needs to be backported, then the PR must be tested against the base branch and the earliest backport release branch and provide tested image version on these two branches. For example, if the PR is requested for master, 202211 and 202012, then the requester needs to provide test results on master and 202012.
-->

#### Which release branch to backport (provide reason below if selected)

<!--
- Note we only backport fixes to a release branch, *not* features!
- Please also provide a reason for the backporting below.
- e.g.
- [x] 202006
-->

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

<!--
- Please provide tested image version
- e.g.
- [x] 20201231.100
-->

- [ ] <!-- image version 1 -->
- [ ] <!-- image version 2 -->

#### Description for the changelog
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->

<!--
 Ensure to add label/tag for the feature raised. example - PR#2174 under sonic-utilities repo. where, Generic Config and Update feature has been labelled as GCU.
-->

#### Link to config_db schema for YANG module changes
<!--
Provide a link to config_db schema for the table for which YANG model
is defined
Link should point to correct section on https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-yang-models/doc/Configuration.md
-->

#### A picture of a cute animal (not mandatory but encouraged)

FengPan-Frank reviewed (commented):
please update https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/doc/Configuration.md, and https://github.com/sonic-net/sonic-buildimage/tree/master/src/sonic-yang-models/tests together.

SerhiyBoikoPLV commented:
> please update https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/doc/Configuration.md, and https://github.com/sonic-net/sonic-buildimage/tree/master/src/sonic-yang-models/tests together.

updated Configuration.md, added tests for poe_port

FengPan-Frank reviewed (approved):
LGTM

mssonicbld commented:
/azp run Azure.sonic-buildimage

Cahb commented:
Rebased on current master and pushed. This is now the only place the yang model changes live. Reviving this, open to re-review.

Cahb commented:
POE_PORT yang model is required for PoE CLI validation.

HLD: https://github.com/sonic-net/SONiC/pull/1631

Related PRs:
- https://github.com/sonic-net/sonic-buildimage/pull/19636
- https://github.com/sonic-net/sonic-utilities/pull/4819
